Fix -Wimplicit-function-declaration error on sigblock() with gcc 14.

```
sig_block.c: In function 'sig_block':
sig_block.c:15:3: error: implicit declaration of function 'sigblock'; did you
mean 'sig_block'? [-Wimplicit-function-declaration]
   15 |   sigblock(1 << (sig - 1));
      |   ^~~~~~~~
      |   sig_block
sig_block.c: In function 'sig_unblock':
sig_block.c:27:3: error: implicit declaration of function 'sigsetmask'; did you
mean 'sigset'? [-Wimplicit-function-declaration]
   27 |   sigsetmask(sigsetmask(~0) & ~(1 << (sig - 1)));
      |   ^~~~~~~~~~
      |   sigset
```

--- a/sig_block.c
+++ b/sig_block.c
@@ -6,35 +6,23 @@

 void sig_block(int sig)
 {
-#ifdef HASSIGPROCMASK
   sigset_t ss;
   sigemptyset(&ss);
   sigaddset(&ss,sig);
   sigprocmask(SIG_BLOCK,&ss,(sigset_t *) 0);
-#else
-  sigblock(1 << (sig - 1));
-#endif
 }

 void sig_unblock(int sig)
 {
-#ifdef HASSIGPROCMASK
   sigset_t ss;
   sigemptyset(&ss);
   sigaddset(&ss,sig);
   sigprocmask(SIG_UNBLOCK,&ss,(sigset_t *) 0);
-#else
-  sigsetmask(sigsetmask(~0) & ~(1 << (sig - 1)));
-#endif
 }

 void sig_blocknone(void)
 {
-#ifdef HASSIGPROCMASK
   sigset_t ss;
   sigemptyset(&ss);
   sigprocmask(SIG_SETMASK,&ss,(sigset_t *) 0);
-#else
-  sigsetmask(0);
-#endif
 }
